jwst/pipeline/calwebb_image3.py

```diff
@@ -69,36 +67,65 @@ def process(self, input_data):
 
         self.source_catalog.save_results = self.save_results
 
-        with datamodels.open(input_data, asn_exptypes=asn_exptypes) as input_models:
+        # Only load science members from input ASN;
+        # background and target-acq members are not needed.
+        input_models = self._load_input_as_library(input_data)
+
+        if (self.output_file is None) and hasattr(input_models.asn["products"][0], "name"):
```

Comment:
```suggestion
        if (self.output_file is None) and "name" in input_models.asn["products"][0]:
```
The `hasattr` replaced here never evaluated to `True` and is contributing to the output naming issues. Applying this suggestion allows removing the change in `source_catalog_step` and (running locally) allows `jwst/regtest/test_nircam_image.py::test_nircam_image_stage3_catalog` to pass.

Reply:
I changed this, then removed the setting of `self.output_file` in both `ResampleStep` and `SourceCatalogStep`, and started regression tests for just `test_nircam_image.py` and `test_miri_image.py` [here](https://github.com/spacetelescope/RegressionTests/actions/runs/10530611112).